### PR TITLE
Added HQ dev FAQ

### PR DESCRIPTION
**Pull request by orangejenny**
Docs only.

Based on https://docs.google.com/document/d/1LrosfV6tuTOhiwOsZ5BUxU5UruSZPMLtQsQ9WVtjMWs/edit and related conversations. This is a little duplicative but hopefully not very. The main point is to have a document separate from the dev setup guide, which is still fairly long, for the kinds of issues that you don't typically run into until you're a few days/weeks past initial setup, and to frame these issues in the ways new devs tend to describe them.

### CODE REVIEW

File: DEV_FAQ.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# Developer FAQ
+
+This is a starting point for troubleshooting issues that frequently occur once your local environment
+is set up and you've started working on a significant code change.
+
+## General Advice
+
+- Use the developers section of the [Dimagi Forum](https://forum.dimagi.com/) for help with issues.
+- Some features are limited to specific software plans. Visit Project Settings > Internal Subscription Management
```
Comment (esoergel) [inline DEV_FAQ.md:9]:
```suggestion
- Some features are limited to specific software plans. Use the `make_domain_enterprise_level` command
```

**Review by esoergel — APPROVED** [review]
I'm sure this will prove very useful!

File: DEV_FAQ.md (hunk 1 above)
Comment (orangejenny) [reply DEV_FAQ.md:9]: nice, I didn't know about that

**Comment by orangejenny** [comment]
killed build

File: DEV_FAQ.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+depending on how much you're working with ES data.
+
+As described in the [dev setup guide](https://github.com/dimagi/commcare-hq/blob/master/DEV_SETUP.md#running-commcare-hq), you can
+use `run_ptop` to keep ES continually up to date.
+
+Alternatively, you can run `ptop_reindexer_v2` as needed to sync individual indexes. `ptop_reindexer_v2` works on one index at a time.
+Running it without any arguments will list the available indexes. The indexes you're most likely to use:
++ `sql-form` and `sql-case` populate form-based and case-based reports and exports
++ `user` and `group` may be useful, depending on what you're working on
++ `case_search` populates the Case list Explorer and Explore Case Data reports, as well as the case search/claim feature
```
Comment (proteusvacuum) [inline DEV_FAQ.md:78]: "for those domains with the `CASE_LIST_EXPLORER`, `EXPLORE_CASE_DATA` or `SEARCH_CLAIM` feature flags enabled"

**Review by proteusvacuum — COMMENTED** [review]
Awesome!

File: DEV_FAQ.md (hunk 2 above)
Comment (orangejenny) [reply DEV_FAQ.md:78]: good point, https://github.com/dimagi/commcare-hq/pull/29895

**Comment by orangejenny** [comment]
FYI @Robert-Costello @stephherbers @gherceg if there's anything you can think of to add, please do.

**Pull request merged**